dali/core/mm/default_resources.cc

```diff
@@ -262,10 +263,10 @@ const std::shared_ptr<pinned_async_resource> &ShareDefaultResourceImpl<memory_ki
     std::lock_guard<std::mutex> lock(g_resources.mtx);
     if (!g_resources.pinned_async) {
       static CUDARTLoader init_cuda;  // force initialization of CUDA before creating the resource
+      g_resources.pinned_async = CreateDefaultPinnedResource();
```

Comment:
What difference does this change make?

Reply:
It fixes static object lifetime issues. If there are any `static` objects created in `CreateDefaultPinnedResource`, they will be alive when `cleanup` object is being destroyed.
For example, if `CreateDefaultPinnedResource` calls `CUDAStreamPool::instance()`, it guarantees that the stream pool singleton will be there when the resource is destroyed. When these lines were swapped, the static variables instantiated as side effects of `CreateDefaultPinnedResource` were instantiated after (and therefore released before), `cleanup`.

Reply:
Understood.